clouddriver-azure/src/main/groovy/com/netflix/spinnaker/clouddriver/azure/client/AzureResourceManagerClient.groovy

```diff
@@ -253,26 +252,24 @@ class AzureResourceManagerClient extends AzureBaseClient {
         body
     } catch (CloudException ce) {  //TODO: (masm) move this error handling logic into the operation classes as part of refactoring how we monitor/report deployment operations/errors
       log.error("Azure Deployment Error: ${ce.body.message}")
+      log.error("Deployment Error Details: {}", ce.body.details.dump())
       throw ce
     } catch (Exception e) {
       log.error("Exception occured during deployment ${e.message}")
       throw e
     } finally {
-      log.info("Template for deployment {}: {}", deploymentName, template)
+      logDeploymentTemplate(deploymentName, template, templateParameters)
     }
   }
 
+  static void logDeploymentTemplate(String deploymentName, String template, Map<String, Object> parameters) {
+    log.info("Template for deployment {}: {}", deploymentName, template)
```

Comment:
Particular reason why this isn't one log message (even if it were multi-lined?) ... otherwise there's a chance that another thread will log in the middle and it'll be more difficult to consolidate the statements.